znai-python/src/main/java/org/testingisdocumenting/znai/python/PythonBasedPythonParser.java

```diff
@@ -47,11 +47,26 @@ public PythonCode parse(Path path) {
         write(path);
         String json = read();
 
+        Map<String, Object> parserResponse;
         try {
-            return new PythonCode((List<Map<String, Object>>) JsonUtils.deserializeAsList(json));
+            parserResponse = (Map<String, Object>) JsonUtils.deserializeAsMap(json);
         } catch (Exception e) {
             throw new RuntimeException("can't parse python parser output: " + json, e);
         }
+
+        boolean success = (boolean) parserResponse.get("success");
+        if (!success) {
+            System.out.println(parserResponse.get("error"));
```

Comment:
this text most likely won't get to the preview page overlay. May need to include it to the exception as well.

Reply:
Ah, I'd forgotten about the preview page overlay.  Hmm, for console output I would prefer to not include it in the exception but maybe that's just the simplest for now.